### ISSUE TITLE
Space added in long attachment file name with special characters

### ISSUE THREAD

**Issue by fundo-moretti** [zbateson/mail-mime-parser#240]
As of version 3.x, when parsing a long attachement file name with special characters, a space is added when encoded file name splits over two lines.

Original file name:
```
Finethic_action S_Décompte_Fondation NEXT_31 mars 2017.pdf
```
Encoded file name: 
```
name="=?UTF-8?Q?Finethic=5Faction_S=5FD=C3=A9compte=5FFondation_NEXT=5F31_mars_2017=2Epd?=
=?UTF-8?Q?f?="
```
Decoded file name:
```
Finethic_action S_Décompte_Fondation NEXT_31 mars 2017.pd f
```
If you remove a single character from file name, decoding is correct; I suspect that the extra space comes from an encoding on two lines.

Example with attached files: [Test.txt](https://github.com/user-attachments/files/16472552/Test.txt)

**Comment by zbateson** [comment]
Hi @fundo-moretti  -- this should be good in 'master' now if you have a chance to give it a go.  I'll aim to release in a week or so if nothing else comes up.

Thanks for the detailed report :)

**Comment by zbateson** [comment]
Released in 3.0.3
